src/wd.rs

```diff
@@ -156,3 +160,118 @@ impl<'a> Locator<'a> {
+///
+/// [`Client::get_timeouts()`]: crate::Client::get_timeouts
+/// [`Client::set_timeouts()`]: crate::Client::set_timeouts
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TimeoutConfiguration {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    script: Option<u64>,
+    #[serde(rename = "pageLoad", skip_serializing_if = "Option::is_none")]
+    page_load: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    implicit: Option<u64>,
+}
+
+impl Default for TimeoutConfiguration {
+    fn default() -> Self {
+        TimeoutConfiguration::new(
+            Some(Duration::from_secs(60)),
+            Some(Duration::from_secs(60)),
+            Some(Duration::from_secs(0)),
+        )
+    }
+}
+
+impl TimeoutConfiguration {
+    /// Create new timeout configuration.
+    ///
+    /// The various settings are as follows:
+    /// - script     Determines when to interrupt a script that is being evaluated.
+    ///              Default is 60 seconds.
+    /// - page_load  Provides the timeout limit used to interrupt navigation of the browsing
+    ///              context. Default is 60 seconds.
+    /// - implicit   Gives the timeout of when to abort locating an element. Default is 0 seconds.
+    ///
+    /// NOTE: It is recommended to leave the `implicit` timeout at 0 seconds, because that makes
+    ///       it possible to check for the non-existence of an element without an implicit delay.
+    ///       Also see [`Client::wait()`] for element polling functionality.
+    ///
+    /// [`Client::wait()`]: crate::Client::wait
+    pub fn new(
+        script: Option<Duration>,
+        page_load: Option<Duration>,
+        implicit: Option<Duration>,
+    ) -> Self {
+        TimeoutConfiguration {
+            script: script.map(|x| x.as_millis() as u64),
+            page_load: page_load.map(|x| x.as_millis() as u64),
+            implicit: implicit.map(|x| x.as_millis() as u64),
+        }
+    }
+
+    /// Get the script timeout.
+    pub fn script(&self) -> Option<Duration> {
+        self.script.map(Duration::from_millis)
+    }
+
+    /// Set the script timeout.
+    pub fn set_script(&mut self, timeout: Option<Duration>) {
+        self.script = timeout.map(|x| x.as_millis() as u64);
+    }
+
+    /// Get the page load timeout.
+    pub fn page_load(&self) -> Option<Duration> {
+        self.page_load.map(Duration::from_millis)
+    }
+
+    /// Set the page load timeout.
+    pub fn set_page_load(&mut self, timeout: Option<Duration>) {
+        self.page_load = timeout.map(|x| x.as_millis() as u64);
+    }
+
+    /// Get the implicit wait timeout.
+    pub fn implicit(&self) -> Option<Duration> {
+        self.implicit.map(Duration::from_millis)
+    }
+
+    /// Set the implicit wait timeout.
+    pub fn set_implicit(&mut self, timeout: Option<Duration>) {
+        self.implicit = timeout.map(|x| x.as_millis() as u64);
+    }
+}
+
+impl From<TimeoutsParameters> for TimeoutConfiguration {
```

Comment:
Unfortunately even this leaks the `webdriver` crate as a public part of `fantoccini`'s API since all trait impls are public. You'll need to have a private inherent conversion method instead. We'll also have to do that for the other adapter types — none of the ones that are `pub` can have trait impls related to `webdriver` types. It's awkward, but unfortunately until we get something like [hidden trait implementations](https://github.com/rust-lang/rfcs/pull/2529) that's what's needed. And until we get [private dependencies](https://github.com/rust-lang/rust/issues/44663), it's hard to catch these too :sweat_smile:

Reply:
Agreed. Done.